src/AclRegs.cc

```diff
@@ -108,6 +106,85 @@
 #if SQUID_SNMP
 #include "snmp_core.h"
 #endif
+#include "sbuf/Stream.h"
+
+namespace Acl
+{
+
+/// Constructs a ParameterizedNode-derived ACL (specified as a Parent class).
+/// This template exists to avoid placing a variant of this ACL construction
+/// code in each ParameterizedNode-derived ACL class just to pass through
+/// TypeName and Parameters onto ParameterizedNode (and to add MEMPROXY_CLASS).
+template <class Parent>
+class FinalizedParameterizedNode: public Parent
+{
+    MEMPROXY_CLASS(Acl::FinalizedParameterizedNode<Parent>);
+
+public:
+    using Parameters = typename Parent::Parameters;
+    using Parent::data;
+
+    /// Replaces generic memory allocator label X set by our MEMPROXY_CLASS(X)
+    /// with an admin-friendly label based on the given acltype-like name.
+    /// Normally, our class constructor sets the right allocator label using the
+    /// actlype name, but that algorithm results in unstable and misleading
+    /// labels when the same instantiation of this template class is used for
+    /// _multiple_ acltype names. Calling this method corrects that behavior.
+    /// \prec this method must be called at most once
+    /// \prec if called, this method must be called before the class constructor
+    static void PreferAllocatorLabelPrefix(const char * const suffix)
```

Comment:
If we remove this and related pool relabeling code in FinalizePoolLabel(), we will essentially go back to the situation in the official code, where all ACLStrategized-based objects are allocated under one admin-unfriendly pool name, regardless of their acltype. The number of the corresponding pool lines in mgr:report will change because the ACLStrategized template was parameterized by ACLData-derived MatchType and this FinalizedParameterizedNode template is parameterized by the ACL-derived Parent class.

This and related pool relabeling code produces acltype-specific pool names. I think this minor code addition is worth it (and it will become even simpler when Mem::AllocatorProxy is finally removed). Here are a few related mgr:mem output line prefixes produced be this PR code:

```
acltype=myportname        112 ...
acltype=port              112 ...
acltype=method            112 ...
acltype=ssl::server_name+ 128 ...
acltype=at_step           112 ...
acltype=srcdomain+        112 ...
acltype=req_header        112 ...
acltype=url_regex         112 ...
```